### DISCUSSION TITLE
API Access???

### DISCUSSION THREAD

**Discussion by websmyths** [etsy/open-api#997] [Q&A]
How in the name of all that is sacred do you get access to Etsy API!? I have been trying for 3 weeks. I thought I had it this morning and then it asked for some more information and now I'm back to pending. The first 2 or 3 times I was denied because I didn't have a website, which is understandable, but a website is not the first thing you do when you're developing an app. First, you'd like to see how the development process is working and if you have something viable. 

There is no practice API you can attempt to use before approval. There is nowhere on the "Manage your apps" page to ask a question or seek help. None of the other Help pages apply. I had to google to find this. I have been waiting for more than a week this time. I have worked with other APIs, even just for experience and nothing has been more difficult than to get access to Etsy's API. It seems to me that Etsy is not interested in having someone develop apps for their platform. It is highly discouraging and I am on the verge of giving up.

**Comment by Gareth064** [comment]
Hey @websmyths. Welcome to the Dev community.

It does appear that getting access to the API is getting harder for new devs coming in. Why this is I'm not sure.
You are 100% right that most devs want to just use the API to test that they can indeed build a proper app for it, so you wouldn't have a fully fledge website ready to submit for the API access process.

Best advice we can give you is to email developer@etsy.com and ask them to give you more advice on your App submission.

**Reply by websmyths** [reply]
Thank you!

**Reply by Gareth064** [reply]
If Etsy Devs respond to your email with any useful insight into the process then please tell us here so others can learn

**Comment by websmyths** [comment]
So I got an answer from Etsy if anyone is interested. Maybe this is not the place to share that, if so let me know. So far they are not allowing apps that involve AI or ML for their API. Although they do allow AI art on their platform, which seems at odd with not allowing AI apps, but that's the way it is. Who knows, just speculation, but maybe they are planning their own AI product.

**Reply by Gareth064** [reply]
Thanks for sharing @websmyths 

Yes Etsy's emails are normally footered with text saying not to share their response. But there are instances where what they may tell you, me or someone else in an email, will benefit the community to know.

This is one of them.

Out of curiousity, how did they know your app had AI or ML features. Did you tell them in the description of your App request, or do you have a website which shows off the feature?
